File: q1750/q1750.java
class Solution {
    public int minimumLength(String s) {
        int left = 0;
        int right = s.length() - 1;
        while (left < right) {
            // If either the left / right pointer points to different character, break
            int currChar = s.charAt(left);
            if (currChar != s.charAt(right))
                break;
            // Move the left pointer as much as possible
            while (left <= right && s.charAt(left) == currChar)
                left += 1;
            // Move the right pointer as much as possible
            while (right > left && s.charAt(right) == currChar)
                right -= 1;
        }
        return right - left + 1;
    }
}

File: q1750/q1750.py
class Solution:
    def minimumLength(self, s: str) -> int:
        left = 0
        right = len(s) - 1
        while left < right: 
            currChar = s[left]
            if currChar != s[right]: break
            while left <= right and currChar == s[left]: left += 1
            while right > left and currChar == s[right]: right -= 1
        return right - left + 1